pyk/src/pyk/kcfg.py

```diff
@@ -535,6 +543,21 @@ def create_edge(self, source_id: str, target_id: str, condition: KInner, depth:
         self._edges[source.id][target.id] = edge
         return edge
 
+    def split_node(self, source_id: str, constraints: Iterable[KInner]) -> List[str]:
+
+        source = self.node(source_id)
+
+        def _add_case_edge(_constraint: KInner) -> str:
+            _cterm = CTerm(mlAnd([source.cterm.term, _constraint]))
+            _node = self.get_or_create_node(_cterm)
+            self.create_edge(source.id, _node.id, _constraint, 0)
+            self.add_verified(source.id, _node.id)
+            return _node.id
+
+        branch_node_ids = [_add_case_edge(constraint) for constraint in constraints]
+        self.add_expanded(source.id)
+        return branch_node_ids
+
```

Comment:
Should we have another marker than `expanded` to show a node split as complete or not?

Reply:
Recorded: https://github.com/runtimeverification/k/issues/2801